packages/react-query/src/__tests__/useQuery.test.tsx

```diff
@@ -5916,8 +5920,84 @@ describe('useQuery', () => {
         </div>
       )
     }
+
     const rendered = renderWithClient(queryClient, <Page />)
     await waitFor(() => rendered.getByText('status: success'))
     await waitFor(() => rendered.getByText('data: 1'))
   })
+  it('should reuse same data object reference when queryKey changes back to some cached data', async () => {
+    jest.useFakeTimers()
+    const spy = jest.fn()
+
+    function fetchNumber(id: number) {
+      return new Promise((resolve) => {
+        setTimeout(
+          () =>
+            resolve({
+              numbers: {
+                current: {
+                  id,
+                },
+              },
+            }),
+          1000,
```

Comment:
we usually try to have smaller timeouts in our tests please. Also, we can make this a bit shorter with our `sleep` util:

```
async function fetchNumber(id: number) {
  await sleep(50)
  return { numbers: { current: { id } } }
}
```

/edit: oh, I see you're using fake timers, so the amount of sleep probably doesn't matter. We don't have fake timers anywhere else I think, and in v5 (alpha branch), we've switched to `vitest`, so maybe we could just do a real sleep but for the shorter time?

Reply:
Yeah sure, I will remove the fake timers since they aren't use anywhere else (they work the same in vitest).

The real timeout leads some tests to have a random behavior depending on the local configuration, I encountered some local failing tests where setting a higher sleep value resolves it.